fetch-in-react/my-app/src/User.js

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import UserCard from './UserCard';
+
+export default function User({ userId, onCancel }) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+  const [user, setUser] = useState();
+
+  /* your code here (hint: useEffect) */
+
+  useEffect(() => {
+    async function fetchResponse(userId) {
+      try {
+        const response = await fetch(
+          'https://jsonplaceholder.typicode.com/users/${userId}'
+        );
+        const result = await response.json();
+        setUser(result);
+      } catch (error) {
+        setError(error);
+      } finally {
+        setIsLoading(!isLoading);
```

Comment:
This works as the code now stands, but it is problematic for a few reasons.

The main reason is that it makes `isLoading` a dependency in your `useEffect`. But if you add `isLoading` as a dependency, you will get infinite re-renders. (Trace the code as we've done in class to understand why.) (By the way, you should be getting a lint warning that you are missing that dependency. If you don't get that warning there is something wrong with your VS Code setup. I might be able to help you with it.)

The other problem is that there's really no guarantee that `isLoading` will be `true` when you start executing this effect. The way the code is currently written, you know it will be `true`, but if someone edits the code in the future they may change that, for some unrelated reason. Since you _know_ that at the end of this effect you will no longer be loading, you should just force the value to `false`.